docs/developer/sre-doc.md

```diff
@@ -0,0 +1,327 @@
+
+### Stop and start the Turbinia stack
+
+Turbinia uses [Pub/Sub](https://cloud.google.com/pubsub/docs/overview) to distribute jobs and tasks to server and
+workers. We must first stop the server before stopping the workers. Before
+stopping the server we need to make sure the server is not handling any open
+tasks/requests anymore.
+
+#### Stop infrastructure
+
+*   Check the server logs to see if it's waiting for any tasks to complete:
+
+    *   <code>$ ./update-gcp-infra.sh -i
+        {INSTANCEID}
+        -z {ZONE} -c logs -s</code>
+
+    ---
+
+    **NOTE** 
+
+    If you are not running from your workstation you can remove the `-s`.
+
+    ---
+
+*   If no tasks are open, shutdown the Turbinia infrastructure. The management
+    script will make sure this is done properly.
+
+    *   <code>$ ./update-gcp-infra.sh -i
+        {INSTANCEID}
+        -z {ZONE} -c stop</code>
+
+    ---
+    **NOTE**
+
+    Workers are shut down asynchronously so it will take a while to show
+    them as "STOPPED".
+
+    ---
+
+#### Start infrastructure
+
+*   The management script will make sure startup is done in the correct order:
+
+    *   <code>$ ./update-gcp-infra.sh -i
+       {INSTANCEID}
+        -z {ZONE} -c start</code>
+
+    ---
+
+    **NOTE**
+
+    workers are started up asynchronously so it will take a while to show
+    them as "RUNNING".
+
+    ---
+
+*   Check the server and worker logs to see if they came back up as expected.
+
+    *   <code>$ ./update-gcp-infra.sh -i
+        {INSTANCEID}
+        -z {ZONE} -c logs -s</code>
+
+    ---
+    **NOTE** 
+
+    If you are not running from your workstation you can remove the `-s`.
+
+    ---
+
+### Configuration update
+
+The Turbinia configuration is base64 encoded as a metadata value
+(TURBINIA\_CONF) attached to the GCE ContainerOS instances. This is written to
+disk at boot time and used by the server and worker docker containers. 
+
+Load the new configuration into the Turbinia stack with the following commands:
+
+*   Stop the stack (see [here](#stop-infrastructure)).
+*   <code>$ ./update-gcp-infra.sh -i
+    {INSTANCEID}
+    -z {ZONE} -f [path-to-new-config-file] -c update-config</code>
+*   Start the stack (see [here](#start-infrastructure)).
+
+### Environment variable update
+
+The Turbinia stack (including monitoring components) sets some configuration
+parameters through containerOS environment variables.
+
+*   Stop the stack (see [here](#stop-infrastructure)).
+*   <code>$ ./update-gcp-infra.sh -i
+    {INSTANCEID}
+    -z {ZONE} -k [env-variable-name] -v [env-variable-value] -c
+    update-config</code>
+*   Start the stack (see [here](#start-infrastructure)).
+
+### Docker image update
+
+Turbinia runs as a docker image (separate image for server and worker) that are
+built and stored in the
+[osdfir-registry](https://github.com/forseti-security/osdfir-infrastructure/blob/2c14857a07f35e1e9f59c5a60c8ef237af93f921/modules/turbinia/variables.tf#L52)
```

Comment:
The link here is pointing to the infrastructure github project, but that is separate from the GCP project that builds the docker images.  Since the GCP project is only available to Turbinia developers we can probably just remove this link altogether.